develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown

```diff
@@ -0,0 +1,384 @@
+
+Go ahead and get started with the upgrade process next.
+
+## Upgrading Your Upgrade Process to the New Framework
+
+Follow the steps below to migrate your code to the new framework.
+
+1.  If your application has any dependencies, add a dependency on the 
+    `portal-upgrade` module to your `build.gradle` file:
+    
+        provided group: "com.liferay", name: "com.liferay.portal.upgrade", 
+        version: "2.0.0"
+
+2.  Check your database schema version against your bundle version.
+
+    If the database schema is in a different version than the bundle version, 
+    specify the different versions for the bundle and schema in your `bnd.bnd` 
+    file, using the `require-SchemaVersion` property:
+
+        require-SchemaVersion: 1.0.2
+
+    If no `Required-SchemaVersion` is found, the `Bundle-Version` header will be 
+    used. Now that the build files are configured, you can move on to the 
+    upgrade class next.
+
+3.  Convert your `upgrade` class into an OSGi component, using the `@Component` 
+    token and implement the `UpgradeStepRegistrator` interface, as shown in the
+    example below:
+    
+        @Component(
+                immediate = true, 
+                service = UpgradeStepRegistrator.class
+        )
+        public class CalendarServiceUpgrade implements UpgradeStepRegistrator       
+
+4.  Remove the intermediate classes that wrapped the internal steps, i.e the 
+    `public class UpgradeProcess_1_0_1 extends UpgradeProcess...`.
+    
+    Following the prior upgrade process in 6.2, you had to define the property 
+    `upgrade.processes`, a list of `UpgradeProcesses` representing the different 
+    upgrades for a specific version of your module, in your 
+    `portal-ext.properties`.
+    
+    For instance, the code below shows the previous process for upgrading 
+    Calendar-service module from v1.0.0 to 1.0.1 and then to 1.0.2. 
+    
+        upgrade.processes=
+            com.liferay.calendar.hook.upgrade.UpgradeProcess_1_0_0,
+            com.liferay.calendar.hook.upgrade.UpgradeProcess_1_0_1,
+            com.liferay.calendar.hook.upgrade.UpgradeProcess_1_0_2
+    
+    Each step between versions was represented by a single class extending
+    `UpgradeProcess`, using a method called `doUpgrade`. This method was 
+    responsible for executing the internal steps to update the database to that 
+    concrete version. A method `getThreshold` was provided also to specify the 
+    schema version where the upgrade starts.
+    
+    Whenever you needed another internal step, you added another
+    `upgrade(new UpgradePortletPreferences());` etc. after the existing ones.
+
+    With the new framework, previous type of classes are represented by upgrade 
+    registrations instead.
+
+    Each upgrade is represented by an upgrade registration. An upgrade 
+    registration is an abstraction for the changes you need to apply to the 
+    database from one version to the next one.
```

Comment:
Mike - It would be good to mention that the developer defines the registrations in an override of [method UpgradeStepRegistrator.register(UpgradeStepRegistrator.Registry)](https://docs.liferay.com/portal/7.0/javadocs/modules/apps/foundation/portal/com.liferay.portal.upgrade/com/liferay/portal/upgrade/registry/UpgradeStepRegistrator.html#register(com.liferay.portal.upgrade.registry.UpgradeStepRegistrator.Registry)).